crates/astria-proto/src/native/sequencer/asset.rs

```diff
@@ -0,0 +1,94 @@
+use std::{
+    error::Error,
+    fmt::Display,
+};
+
+/// The default sequencer asset base denomination.
+pub const DEFAULT_NATIVE_ASSET_DENOM: &str = "nria";
+
+/// Returns the default sequencer asset ID.
+#[must_use]
+pub fn default_native_asset_id() -> Id {
+    Denom::from_base_denom(DEFAULT_NATIVE_ASSET_DENOM).id()
+}
+
+/// Represents a denomination of a sequencer asset.
+#[derive(Debug, Clone)]
+pub struct Denom {
+    id: Id,
+
+    /// The base denomination of the asset; ie. the name of
+    /// the smallest unit of the asset.
+    base_denom: String,
+}
+
+impl Denom {
+    #[must_use]
+    pub fn from_base_denom(base_denom: &str) -> Self {
+        let id = Id::from_denom(base_denom);
+
+        Self {
+            id,
+            base_denom: base_denom.to_string(),
+        }
+    }
+
+    #[must_use]
+    pub fn id(&self) -> Id {
```

Comment:
Can you add little doc strings to this that also explain what these things are (i.e. hash of the base denom, etc)